src/components/NewDatePicker/index.js

```diff
@@ -42,17 +49,42 @@ const datePickerDefaultProps = {
     minDate: moment().year(CONST.CALENDAR_PICKER.MIN_YEAR).toDate(),
     maxDate: moment().year(CONST.CALENDAR_PICKER.MAX_YEAR).toDate(),
     value: undefined,
+    shouldSaveDraft: false,
+    formID: '',
 };
 
-function NewDatePicker({containerStyles, defaultValue, disabled, errorText, inputID, isSmallScreenWidth, label, maxDate, minDate, onInputChange, onTouched, placeholder, translate, value}) {
+function NewDatePicker({
+    containerStyles,
+    defaultValue,
+    disabled,
+    errorText,
+    inputID,
+    isSmallScreenWidth,
+    label,
+    maxDate,
+    minDate,
+    onInputChange,
+    onTouched,
+    placeholder,
+    translate,
+    value,
+    shouldSaveDraft,
+    formID,
+}) {
     const [selectedDate, setSelectedDate] = useState(value || defaultValue || undefined);
 
     useEffect(() => {
+        // Value is provided to input via props and onChange never fires. We have to save draft manually.
+        if (shouldSaveDraft && formID !== '') {
```

Comment:
do we need a `shouldSaveDraft` prop? Looks like for this action we need only formID. When we pass shouldSaveDraft but without formID this action still doesn't work. So looks like shouldSaveDraft is not needed

Reply:
I don't remember exactly as @MrMuzyk made this change in his PR, however I believe this is to keep the inputs and `shouldSaveDraft` the same.

Reply:
Its true that `shouldSaveDraft` is not needed. I think it would make sense to cut it and rename `formID` to `formIDToSaveDraft` so it is clear what will happen once we pass it